### PR TITLE
Add logger to `main` package and improve logging

### LINKED ISSUE #85: improve app logging
# Summary

We need improve and standardize format of logs records because current implementation has few issues which can do problem with integration of log collector:

- "Main error" reason for terminate of application is printed with standard fmt.Println() when other errors use zap logger.
   https://github.com/ironcore-dev/libvirt-provider/blob/main/provider/cmd/main.go#L18
- Duplicated "main error" logs:
   ```text
   Error: failed to initialize machine class registry: unable to unmarshal machine classes: error converting YAML to JSON: yaml: line    2: did not find expected alphabetic or numeric character
   failed to initialize machine class registry: unable to unmarshal machine classes: error converting YAML to JSON: yaml: line 2: did     not find expected alphabetic or numeric character
   ```
- We should use production configuration as default. So i recommend disable development level for zap logger as default settings because if you need it, you can use argument --zap-devel
- Disable print of usage for every "main error" which terminated application. We can print usage, if users use wrong argument or required arguments are missing, but it doesn't make sense print usage for application runtime error.

## Motivation

Easier integration log collectors with our application

### PR DESCRIPTION
**Pull request by lukasfrank**
# Proposed Changes

Improved logging
- Replaced main error logging from `fmt.Println` to logger
- Removed redundant logging by logger and cobra
- Disabled usage help once the flags are parsed 

Fixes #85

### CODE REVIEW

File: provider/cmd/main.go [hunk 1]
```diff
@@ -13,9 +12,10 @@ import (
 
 func main() {
 	ctx := ctrl.SetupSignalHandler()
+	mainLog := ctrl.Log.WithName("main")
```
Comment (afritzler) [inline provider/cmd/main.go:15]: Just use `log` here as it is clear from the context what this logger is being used for.

**Review by afritzler — CHANGES_REQUESTED** [review]

File: provider/cmd/main.go (hunk 1 above)
Comment (lukasfrank) [reply provider/cmd/main.go]: fixed

**Review by afritzler — DISMISSED** [review]

File: provider/cmd/app/app.go [hunk 2]
```diff
@@ -372,7 +391,7 @@ func runGRPCServer(ctx context.Context, setupLog logr.Logger, log logr.Logger, s
 
 	grpcSrv := grpc.NewServer(
 		grpc.ChainUnaryInterceptor(
-			commongrpc.InjectLogger(log),
+			commongrpc.InjectLogger(log.WithName("grpc-server")),
```
Comment (lukas016) [inline provider/cmd/app/app.go:394]: optional: why grpc? it is iri protocol server, which implement your iri protocol. grpc is used technology.

File: provider/cmd/main.go [hunk 3]
```diff
@@ -13,9 +12,10 @@ import (
 
 func main() {
 	ctx := ctrl.SetupSignalHandler()
+	log := ctrl.Log.WithName("main")
 
 	if err := app.Command().ExecuteContext(ctx); err != nil {
-		fmt.Println(err.Error())
+		log.V(1).Error(err, "Error running libvirt provider")
```
Comment (lukas016) [inline provider/cmd/main.go:18]:
```suggestion
		log.V(1).Error(err, "error running libvirt provider")
```

i prefer same format of error messages.

File: provider/cmd/app/app.go (hunk 2 above)
Comment (lukasfrank) [reply provider/cmd/app/app.go]: fixed

File: provider/cmd/main.go (hunk 3 above)
Comment (lukasfrank) [reply provider/cmd/main.go]: fixed

**Review by lukas016 — APPROVED** [review]

**Pull request merged**